library/patient.inc.php

```diff
@@ -1822,3 +1822,17 @@ function updateDupScore($pid)
     );
     return $dupscore;
 }
+
+function get_unallocated_payment_id($pid)
+{
+    $query = "SELECT a.session_id " .
+        "FROM ar_session AS a " .
+        "WHERE a.patient_id = ? AND " .
+        "a.adjustment_code = 'pre_payment' AND a.closed = 0 ORDER BY a.check_date ASC LIMIT 1";
```

Comment:
Does check date get populated if a cash payment which occurs in some of your use cases?
Also it drives me nut when an alias is used when not necessary i.e ar_session as a!:) It slows down fetch. Just a personal note. Do what works...

Reply:
oh never mind check_date isn't pert of query, sorry!